app/controllers/support_controller.rb

```diff
@@ -2,100 +2,77 @@ class SupportController < ApplicationController
   invisible_captcha only: [:create], on_spam: :redirect_to_root
 
   def index
-    setup_context
+    @form = ContactForm.new(tags: support_form_params.fetch(:tags, []), dossier_id: dossier&.id)
+    @form.user = current_user
   end
 
   def admin
-    setup_context_admin
+    @form = ContactForm.new(tags: support_form_params.fetch(:tags, []), for_admin: true)
+    @form.user = current_user
   end
 
   def create
-    if direct_message? && create_commentaire
-      flash.notice = "Votre message a été envoyé sur la messagerie de votre dossier."
+    if direct_message?
+      create_commentaire!
+      flash.notice = t('.direct_message_sent')
 
       redirect_to messagerie_dossier_path(dossier)
       return
     end
 
-    create_conversation_later
-    flash.notice = "Votre message a été envoyé."
+    form_params = support_form_params
+    @form = ContactForm.new(form_params.except(:piece_jointe))
+    @form.piece_jointe.attach(form_params[:piece_jointe]) if form_params[:piece_jointe].present?
```

Comment:
je suis surpris que tu aies ce genre de danse à faire, je m'attendais à qqchose de plus built in